lib/oidc/util/browser.ts

```diff
@@ -80,3 +81,36 @@ export function addPostMessageListener(sdk: OktaAuthOAuthInterface, timeout, sta
       removeListener(window, 'message', responseHandler);
     });
 }
+
+export function addIDPPopupLisenter (
+  sdk: OktaAuthOAuthInterface,
+  timeout: number | undefined,
+  channel: BroadcastChannel,
+  state: string
+) {
+  let timeoutId;
+
+  const promise = new Promise((resolve, reject) => {
+    channel.onmessage = (event) => {
+      if (!event.isTrusted || !event.data) {
+        return;
+      }
+
+      if (typeof event.data === 'string') {
+        return resolve({ state, code: event.data });
+      }
+
+      reject(new AuthSdkError('Unable to complete auth code exchange'));
```

Comment:
For the final PR we should consider adding more detail about why, e.g. if event.data wasn't a string, mention that